### ISSUE TITLE
unSelectAll on a multiple select does not work on Safari

### ISSUE THREAD

**Issue by davidearl** [HemantNegi/jquery.sumoselect#299]
(...and I'm puzzled why it seems to work on Chrome)

`$('select.SlectBox')[0].sumo.unSelectAll()
`
throws an error on Safari. This seems to be because the `find` currently at line 516 of jquery.sumoselect.js uses ':selected' which is not allowed for multiple selects, should be ':checked'. (Why couldn't they be consistent about this across all UI elements?!)

I worked around it with 
`$('select.SlectBox').val([])[0].sumo.reload()
`

```
setText: function () {   
                    var O = this;  
                    O.placeholder = "";
                    if (O.is_multi) {
                        var sels = O.E.find(':selected').not(':disabled'); //selected options.
 //                                           ^ here
```

**Comment by PlayToDominate** [comment]
@davidearl how/where did you apply your workaround? I'm having the same issue for iPad both on selectall and unselectall

**Comment by davidearl** [comment]
I didn't change the SumoSelect code at all, just used the alternative code I gave, which for deselect sets the original select to nothing selected and then reloads the sumoselect state.

However, I believe if you change selected to checked in the SumoSelect code, it should then work. The line number may have changed since I wrote that.

For select all I guess you could workaround by getting and array of all the values of the original select using ```
.children("option").each(...)``` , setting that as the select cal and reloading.

**Comment by Zenoo** [comment]
This should be fixed by c759fd384cf4084d5309edb29427540f01290c6f
